### PR TITLE
Use jackson-dataformat-csv for csv/ssv/tsv handling

### PR DESCRIPTION
**Pull request by soenkeliebau**
When using the connector we ran into a few limitations of the current csv, ssv and tsv parsing that caused the connector to fail for those cases.

The specific issue we encountered was with the separator occurring in fields or headers which where then quoted to make it "legal".

An example of this is:

```
header1,header2,header3
value1,"value, 2",value3
```

Since the current parsing logic splits on all occurrences of` , `this would break, as the second line looks like it contains an additional field.

Since the project is already using jackson, which these days also supports csv parsing, it might make sense to extend the code a bit, in order to properly handle csv,ssv and tsv quoting and similar shenaningans.

This PR adds a dependency on jackson-dataformat-csv to add csv parsing functionality and replaces the current '.split(separator)' way of handling csv/ssv/tsv with the jackson parser.

**Open Questions**
1. Trimming
One thing that I noticed when implementing this is that tests failed originally, because my first implementation didn't trim whitespaces.
Personally I'd consider not trimming whitespaces to be the better behavior, as they may be there for a reason and it shouldn't be the job of this connector to guess whether they were intentional or not.
Is there a historic reason for the trimming to be there and do people rely on this functionality, or is this something that might be changed?

2. Multi Character Separators
Jackson only supports single char separators, but we currently take a String for the separator, do we envision actually supporting multi-char separators at some point? 
If not, would it make sense to change this to a char now, as this implementation currently ignores anything beyond the first character of the separator.

### CODE REVIEW

**Comment by soenkeliebau** [comment]
ping @nineinchnick & @ebyhr  
I cannot request reviews directly, sorry.

File: src/test/resources/example-data/quoted_fields_with_separator.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+header_1,header_2,"weird, but valid header 3", header_4_with_extra_whitespace
+test,2,3,4
+"test,test,test,test",3,3,5
+" even weirder, but still valid, value with extra whitespaces that remain due to quoting /  ",1,2,3
```
Comment (nineinchnick) [inline src/test/resources/example-data/quoted_fields_with_separator.csv:4]: Would it also work with newlines in quoted values?

File: src/main/java/org/ebyhr/trino/storage/operator/CsvPlugin.java [hunk 2]
```diff
@@ -32,21 +34,24 @@ public class CsvPlugin
...
-        Splitter splitter = Splitter.on(delimiter).trimResults();
-
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(streamProvider.apply(path)))) {
-            List<String> fields = splitter.splitToList(reader.readLine());
-            return fields.stream()
-                    .map(field -> new StorageColumnHandle(field, VARCHAR))
+        try {
+            // Read the first line and use the values as column names
+            MappingIterator<List<String>> it = this.mapper.readerFor(List.class).with(schema).readValues(streamProvider.apply(path));
+            return it.next().stream().map(field -> new StorageColumnHandle(field, VARCHAR))
```
Comment (nineinchnick) [inline src/main/java/org/ebyhr/trino/storage/operator/CsvPlugin.java:54]: Make it a tiny bit easier to read.
```suggestion
            MappingIterator<List<String>> it = this.mapper.readerFor(List.class).with(schema).readValues(streamProvider.apply(path));
            List<String> fields = it.next();
            return fields.stream()
                    .map(field -> new StorageColumnHandle(field, VARCHAR))
```

**Review by nineinchnick — APPROVED** [review]

File: src/test/resources/example-data/quoted_fields_with_separator.csv (hunk 1 above)
Comment (soenkeliebau) [reply src/test/resources/example-data/quoted_fields_with_separator.csv:4]: It actually would ... I'd consider that to be moving into abomination territory, but it does work :)
Added tests for this in https://github.com/snowlift/trino-storage/pull/244/commits/984316f2fa82431bb264a801fd67131169084f04

File: src/main/java/org/ebyhr/trino/storage/operator/CsvPlugin.java (hunk 2 above)
Comment (soenkeliebau) [reply src/main/java/org/ebyhr/trino/storage/operator/CsvPlugin.java]: Done https://github.com/snowlift/trino-storage/pull/244/commits/24754868b60bffc5a13ed0669a6532565ebc5c47

Comment (nineinchnick) [reply src/main/java/org/ebyhr/trino/storage/operator/CsvPlugin.java]: Thanks, please squash all commits and we can merge this.

**Comment by soenkeliebau** [comment]
Do you have any opinions on the two open questions I mentioned in the PR @nineinchnick ?

**Comment by nineinchnick** [comment]
Ad 1. I agree, but let's preserve the current behavior.
Ad 2. We don't have to worry about multi-char separators.

At some point we might start a discussion on how to extend the table functions to make the readers more configurable, that is add some arguments to override stuff like separators, quotes, trimming etc.

**Comment by soenkeliebau** [comment]
> At some point we might start a discussion on how to extend the table functions to make the readers more configurable, that is add some arguments to override stuff like separators, quotes, trimming etc.

I agree, but that day is not today I think :)

**Comment by soenkeliebau** [comment]
Changed the implementation a bit to take a char, so we don't give the impression of allowing multi-character separators.

**Comment by nineinchnick** [comment]
Thanks! Please squash the three first commits, except `Changed implementation of CsvPlugin to only allow single-char separators.`

**Comment by soenkeliebau** [comment]
done

**Comment by nineinchnick** [comment]
Sorry, one last thing - can you make the title of the first commit shorter? Something like "Support quotes values in CSV files"

**Comment by soenkeliebau** [comment]
done

File: pom.xml [hunk 3]
```diff
@@ -56,6 +56,7 @@
         <dep.guice.version>7.0.0</dep.guice.version>
         <dep.errorprone.version>2.23.0</dep.errorprone.version>
         <dep.jackson.version>2.15.3</dep.jackson.version>
+        <dep.jackson-dataformat-csv.version>2.15.0</dep.jackson-dataformat-csv.version>
```
Comment (nineinchnick) [inline pom.xml:59]: The latest version for this is 2.15.3, which I'd assume matches all other Jackson components. Can we just use `dep.jackson.version` for this?

**Review by nineinchnick — APPROVED** [review]
One last comment and this is ready to be merged.

File: pom.xml (hunk 3 above)
Comment (soenkeliebau) [reply pom.xml:59]: Ups, absolutely. I went looking at git tags when I started on this and could only find a 2.15.0 for the csv part, hence I split that and figured they are doing separate bugfix lines.
But you are of course right, there is a 2.15.3. I'll change and rebase.

**Review by nineinchnick — APPROVED** [review]

**Comment by nineinchnick** [comment]
@soenkeliebau I just released 432 with this change.

**Comment by soenkeliebau** [comment]
thanks!!

**Pull request merged**